File: test/twtest.pl
#!/usr/local/bin/perl

use strict;
use warnings;
use utf8;

use Net::Twitter::Lite;
use Data::Dumper;

=pod
twitterでOAuth認証を使ってつぶやくだけのサンプル
事前に http://dev.twitter.com/ で app として登録しておくこと
=cut

# http://dev.twitter.com/apps/XXXXXX で取得できるやつ
my %CONSUMER_TOKENS = (
    consumer_key    => 'oVHQOYjXkfrEOGEVdRosQ',
    consumer_secret => '5Z0zGHDWqBshT1nWa3wcCB7fx69kH7cNExPPdHAGR8',
);

# http://dev.twitter.com/apps/XXXXXX/my_token で取得できるやつ
my $ACCESS_TOKEN        = '207520259-a5z3WtxYG807hJGT1Ulat1GUcqolTX2dUPF0oVZT';
my $ACCESS_TOKEN_SECRET = '8c8P03bhKbOzwSnPYAxYBJZ6Hm9dscZ4Vwrffl356Pg';

# constructs a "Net::Twitter::Lite" object
my $t = Net::Twitter::Lite->new(%CONSUMER_TOKENS);

# トークンをセットする
$t->access_token($ACCESS_TOKEN);
$t->access_token_secret($ACCESS_TOKEN_SECRET);

# 投稿
my $status = $t->update({ status => 'Perlで投稿テスト' });

# 投稿した結果をダンプ
print Dumper $status;

File: test/twtest.php
<?php
// twitteroauth.phpを読み込む。パスはあなたが置いた適切な場所に変更してください
require_once("twitteroauth.php");

// Consumer keyの値
$consumer_key = "oVHQOYjXkfrEOGEVdRosQ";
// Consumer secretの値
$consumer_secret = "5Z0zGHDWqBshT1nWa3wcCB7fx69kH7cNExPPdHAGR8";
// Access Tokenの値
$access_token = "207520259-a5z3WtxYG807hJGT1Ulat1GUcqolTX2dUPF0oVZT";
// Access Token Secretの値
$access_token_secret = "8c8P03bhKbOzwSnPYAxYBJZ6Hm9dscZ4Vwrffl356Pg";

// OAuthオブジェクト生成
$to = new TwitterOAuth($consumer_key,$consumer_secret,$access_token,$access_token_secret);

// TwitterへPOSTする。パラメーターは配列に格納する
// in_reply_to_status_idを指定するのならば array("status"=>"@hogehoge reply","in_reply_to_status_id"=>"0000000000"); とする。
$req = $to->OAuthRequest("https://twitter.com/statuses/update.xml","POST",array("status"=>"OAuth経由のポストテスト"));
// TwitterへPOSTするときのパラメーターなど詳しい情報はTwitterのAPI仕様書を参照してください

header("Content-Type: application/xml");
echo $req;
?>
